src/app/shared/services/screen-orientation/screen-orientation.service.ts

```diff
@@ -1,21 +1,47 @@
-import { Injectable } from "@angular/core";
-import { SyncServiceBase } from "../syncService.base";
+import { effect, Injectable } from "@angular/core";
 import { ScreenOrientation, OrientationLockType } from "@capacitor/screen-orientation";
 import { TemplateActionRegistry } from "../../components/template/services/instance/template-action.registry";
+import { Capacitor } from "@capacitor/core";
+import { TemplateMetadataService } from "../../components/template/services/template-metadata.service";
+import { SyncServiceBase } from "../syncService.base";
+
+// Supported orientation types
+const ORIENTATION_TYPES = ["portrait", "landscape"] as const;
 
-const ORIENTATION_TYPES: OrientationLockType[] = ["portrait", "landscape"];
+type IOrientationType = (typeof ORIENTATION_TYPES)[number];
 
 @Injectable({
   providedIn: "root",
 })
 export class ScreenOrientationService extends SyncServiceBase {
-  constructor(private templateActionRegistry: TemplateActionRegistry) {
+  private enabled: boolean;
+
+  constructor(
+    private templateActionRegistry: TemplateActionRegistry,
+    private templateMetadataService: TemplateMetadataService
+  ) {
     super("Screen Orientation Service");
+
+    // TODO: expose a property at deployment config level to enable "landscape_mode" to avoid unnecessary checks
+    // AND/OR: check on init if any templates actually use screen orientation metadata?
+    this.enabled = Capacitor.isNativePlatform();
+
+    if (this.enabled) {
+      effect(() => {
+        const targetOrientation =
+          this.templateMetadataService.parameterList().orientation || "portrait";
```

Comment:
nit(blocking): this will force all templates without parameter to lock portrait mode, which might be undesirable for users on tablet devices who prefer to keep in landscape.